### PR TITLE
Added the Size Measurement Block

### PR DESCRIPTION
**Pull request by chandlersupple**
# Description

Added the Size Measurement block. The Size Measurement block measures the dimensions of an object using a reference object with known dimensions. In addition, I added `test_size_measurement.py` with unit tests.

Screenshot 2024-10-02 at 2 03 29 PM

## Type of change

Please delete options that are not relevant.

-   [X] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?

Passed all existing and new tests.

## Any specific deployment considerations

N/A

## Docs

N/A

### CODE REVIEW

File: inference/core/workflows/core_steps/transformations/size_measurement/v1.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+    def run(
+        self,
+        reference_predictions: Batch[sv.Detections],
+        object_predictions: Batch[sv.Detections],
+        reference_dimensions: Union[str, Tuple[float, float]],
+    ) -> BlockResult:
+        result = []
+        ref_width_actual, ref_height_actual = (
+            SizeMeasurementManifest.parse_reference_dimensions(reference_dimensions)
```
Comment (PawelPeczek-Roboflow) [inline inference/core/workflows/core_steps/transformations/size_measurement/v1.py:131]: If I understand correctly,  `@validator("reference_dimensions", pre=True)` should make the transformation while manifest parsing and you would only be able to get `Tuple[float, float]` as `reference_dimensions`, but I may be mistaken

File: inference/core/workflows/core_steps/transformations/size_measurement/v1.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        )
+
+        for ref_detections, obj_detections in zip(
+            reference_predictions, object_predictions
+        ):
+            if ref_detections is None or obj_detections is None:
```
Comment (PawelPeczek-Roboflow) [inline inference/core/workflows/core_steps/transformations/size_measurement/v1.py:137]: should not happen if Block do not declare to accept empty inputs

File: inference/core/workflows/core_steps/transformations/size_measurement/v1.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    object_predictions: StepOutputSelector(
+        kind=[
+            INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            OBJECT_DETECTION_PREDICTION_KIND,
+        ]
+    ) = Field(
+        description="Predictions from the model that detects the object to measure",
+        examples=["$segmentation.object_predictions"],
+    )
+    reference_dimensions: Union[str, Tuple[float, float]] = Field(
```
Comment (PawelPeczek-Roboflow) [inline inference/core/workflows/core_steps/transformations/size_measurement/v1.py:67]: For practical usefulness I suggest letting people to inject reference dimensions through input parameter - that would make it impossible to always verify the input validity at compile time, but you can still validate and raise error in `run(...)` method once someone injects input parameter in invalid form

File: inference/core/workflows/core_steps/transformations/size_measurement/v1.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        json_schema_extra={
+            "name": "Size Measurement",
+            "version": "v1",
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "transformation",
+        }
+    )
+    type: Literal[f"{TYPE}", "SizeMeasurement"]
```
Comment (PawelPeczek-Roboflow) [inline inference/core/workflows/core_steps/transformations/size_measurement/v1.py:48]: Please just put `roboflow_core/size_measurement@v1` here, without `TYPE` var and size measurement alias

File: inference/core/workflows/core_steps/transformations/size_measurement/v1.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    WorkflowBlockManifest,
+)
+
+OUTPUT_KEY: str = "dimensions"
+TYPE: str = "roboflow_core/size_measurement@v1"
+SHORT_DESCRIPTION = (
+    "Measure the dimensions of objects in relation to a reference object."
+)
+LONG_DESCRIPTION = """
+The `SizeMeasurementBlock` is a transformer block designed to measure the dimensions of objects
```
Comment (PawelPeczek-Roboflow) [inline inference/core/workflows/core_steps/transformations/size_measurement/v1.py:30]: The way on how the block works is hard to be understood from description - lack of mention that `reference_predictions` may provide multiple boxes, but the most confident one will be selected - we shall also suggest to combine the block with "filter + potentially continue if" when different behaviour is expected.

**Review by grzegorz-roboflow — APPROVED** [review]

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Pull request merged**